test/e2e/specs/editor/various/embedding.spec.js

```diff
@@ -301,10 +305,10 @@ class EmbedUtils {
 	async insertEmbed( url ) {
 		await test.step( `Inserting embed ${ url }`, async () => {
 			await this.#editor.insertBlock( { name: 'core/embed' } );
-			await this.#editor.canvas
-				.getByRole( 'textbox', { name: 'Embed URL' } )
-				.last()
-				.fill( url );
+			// Do not use `fill` here, it's not how the user interacts with the
+			// block.
+			this.#pageUtils.setClipboardData( { plainText: url } );
+			await this.#pageUtils.pressKeys( 'primary+v' );
```

Comment:
@kevin940726 reminded me that this currently doesn't work on regular elements. See https://github.com/WordPress/gutenberg/pull/55030.

Reply:
Hm, interesting. I fixed it a bit differently here: 3bed7b6ff4c57cb0032f9333fafa27533e790c29

Reply:
@kevin940726, do you mind having a look?

Reply:
I'm afraid it's not that simple. It's a cross-browser compatibility issue too. PR #55030 should work fine but it's too complicated to review. I haven't had the time to refactor it to a simpler form. I think it's okay to omit this change and just add the `test.skip` and a comment for now.

Reply:
Why skip? It actually tests a regression? Could you elaborate on why it's not that simple? Currently all our copy/paste tests are emulated and we manually fill in data, so how is this different?

This is a pretty bad regression, so we should get at least the fix in asap.

Reply:
Playwright doesn't handle native copy/paste events nicely at the moment. There are subtle differences between different browsers (chromium, firefox, webkit), and different OSes (linux, mac, windows). What makes this even more complicated is that sometimes we have to emulate it by manually dispatching events, but some other times we have to use the native events. A proper "fix" might be maintaining a table of these factors and keeping them updated as we update Playwright or our OSes.

I totally agree we should get the fix in. However, the changes in the e2e utils might not work perfectly in all cases. What I'm suggesting is omitting the changes in e2e utils and adding an e2e test marked as `test.skip` so that we can cover this case while refactoring #55030.

I'm not entirely opposed to merging the e2e utils changes either, just that we might have to refactor it again sometimes. And I personally prefer to do that at once to avoid unintentional breakage 😅.

Reply:
I think it's fine if we have to refactor it again. To me the most important thing is that we have good e2e test coverage.

Reply:
I don't mind merging this as is 👍. I just think that the test might not have good coverage after all if we're "designing" the test in a way that it should pass. I guess it's hard to argue against not having the test at all though 😆.

Reply:
Not sure I agree there. We're "designing" all copy/paste tests to pass then? We're actually manually emulating everything, this is no different? For example for copy/cut, we're manually setting a clipboard instead of the browser doing the default behaviour. The test assures that default behaviour is not prevented, and we emulate the default behaviour.

Reply:
Yeah, I think I chose the wrong words😅. I couldn't find a better word. What I mean is that for this particular use case, we might not want to intercept what Playwright already does. It's to prevent future breakage of our tests when the bundled browsers update. The more robust way might be to feature-detect it in our utils, but even that is hard to do right. I hope that this convo didn't sound discouraging or blaming! I'm just not sure what the best approach is at the time, but I'm positive that we can find one! ❤️

Reply:
Not at all! ❤️ Just saying I prefer to err on the side of adding a test, even if it's not perfect, but I'm open to your util changes as a follow-up